src/plugins/audioplayers/plugin.c

```diff
@@ -0,0 +1,211 @@
+        bool looping = strstr(releaseMode, "loop") != NULL;
+        AudioPlayer_SetLooping(player, looping);
+        result = 1;
+    } else if (strcmp(method, "setPlayerMode") == 0) {
+        // TODO check support for low latency mode:
+        // https://gstreamer.freedesktop.org/documentation/additional/design/latency.html?gi-language=c
+        result = 1;
+    } else {
+        return platch_respond_not_implemented(responsehandle);
+    }
+
+    return platch_respond_success_std(responsehandle, &STDINT64(result));
+}
+
+static int on_global_method_call(char* channel, struct platch_obj* object, FlutterPlatformMessageResponseHandle* responsehandle)
+{
+    const char* method;
+
+    (void)responsehandle;
+    (void)channel;
+
+    method = object->method;
+
+    return platch_respond(
+      responsehandle,
+      &(struct platch_obj){ .codec = kStandardMethodCallResponse, .success = true, .std_result = { .type = kStdTrue } });
+}
+
+enum plugin_init_result audioplayers_plugin_init(struct flutterpi* flutterpi, void** userdata_out)
+{
+    (void)userdata_out;
+    int ok;
+    plugin.flutterpi = flutterpi;
+    plugin.initialized = false;
+
+    ok = cpset_init(&plugin.players, CPSET_DEFAULT_MAX_SIZE);
+    if (ok != 0)
+        return kError_PluginInitResult;
+
+    ok = plugin_registry_set_receiver(AUDIOPLAYERS_GLOBAL_CHANNEL, kStandardMethodCall, on_global_method_call);
+    if (ok != 0) {
+        goto fail_deinit_cpset;
+    }
+
+    ok = plugin_registry_set_receiver(AUDIOPLAYERS_LOCAL_CHANNEL, kStandardMethodCall, on_local_method_call);
+    if (ok != 0) {
+        goto fail_remove_global_receiver;
+    }
+
+    return kInitialized_PluginInitResult;
+
+fail_remove_global_receiver:
+    plugin_registry_remove_receiver(AUDIOPLAYERS_GLOBAL_CHANNEL);
+
+fail_deinit_cpset:
+    cpset_deinit(&plugin.players);
+
+    return kError_PluginInitResult;
+}
+
+void audioplayers_plugin_deinit(struct flutterpi* flutterpi, void* userdata)
+{
+    (void)flutterpi;
+    (void)userdata;
+    plugin_registry_remove_receiver(AUDIOPLAYERS_GLOBAL_CHANNEL);
+    plugin_registry_remove_receiver(AUDIOPLAYERS_LOCAL_CHANNEL);
+
+    AudioPlayer* ptr;
+    for_each_pointer_in_cpset(&plugin.players, ptr)
+    {
+        AudioPlayer_Dispose(ptr);
+    }
+
+    cpset_deinit(&plugin.players);
+}
+
+static AudioPlayer* audioplayers_linux_plugin_get_player(char* playerId, char* mode)
+{
+    (void)mode;
+    AudioPlayer* player = NULL;
+    AudioPlayer* p;
+    for_each_pointer_in_cpset(&plugin.players, p)
+    {
+        if (!strcmp(p->playerId, playerId)) {
+            player = p;
+            break;
+        }
+    }
+    if (player != NULL) {
+        return player;
+    } else {
+        AudioPlayer* player = AudioPlayer_new(playerId, AUDIOPLAYERS_LOCAL_CHANNEL);
+        cpset_put_locked(&plugin.players, player);
+        return player;
+    }
+}
+
+FLUTTERPI_PLUGIN("audioplayers_flutter_pi", audioplayers, audioplayers_plugin_init, audioplayers_plugin_deinit)
```

Comment:
You probably want to stay consistent with the naming of the official audioplayer backends here, but IMO you can leave away the `_flutter_pi` because that's obvious 😄